### PR TITLE
refactor: Use an enum for shutdown values

### PR DESCRIPTION
**Pull request by JelteF**
In #902 I want to introduce a 4th shutdown value. So I started using an
enum for this to improve readibility. I think this refactor from using
integer values to enum values stands on its own. So to make reviewing
easier, and not have it be blocked on adding tests and some discussions
for #902 this PR includes only that refactor.

### CODE REVIEW

File: include/pooler.h [hunk 1]
```diff
@@ -22,6 +22,7 @@ void resume_pooler(void);
 void suspend_pooler(void);
 void per_loop_pooler_maint(void);
 void pooler_tune_accept(bool on);
+void cleanup_tcp_sockets(void);
```
Comment (eulerto) [inline include/pooler.h:25]: It seems unrelated to this change.

File: include/bouncer.h [hunk 2]
```diff
@@ -75,6 +75,12 @@ enum PauseMode {
 	P_SUSPEND = 2		/* wait for buffers to be empty */
 };
 
+enum ShutDownMode {
+	SHUTDOWN_NONE = 0,
+	SHUTDOWN_WAIT_FOR_SERVERS = 1,
+	SHUTDOWN_IMMEDIATE = 2,
+};
```
Comment (eulerto) [inline include/bouncer.h:82]: Do you need the numbers?

**Review by eulerto — APPROVED** [review]

File: include/bouncer.h (hunk 2 above)
Comment (JelteF) [reply include/bouncer.h:84]: The 0 one is necessary, so we can do `if (cf_shutdown) { ... }`. So I left that one and removed the others, because those were indeed not needed.

File: include/pooler.h (hunk 1 above)
Comment (JelteF) [reply include/pooler.h]: Thanks for noticing, that one should be part of #902 only.

File: include/bouncer.h (hunk 2 above)
Comment (eulerto) [reply include/bouncer.h:84]: There is no harm specifying the `= 0` but the spec says it starts with zero if not specified [1]. Since you mention this `if`:

```
if (cf_shutdown != SHUTDOWN_NONE) {
...
```

is my preference.

[1] https://en.cppreference.com/w/c/language/enum

Comment (JelteF) [reply include/bouncer.h:84]: >There is no harm specifying the = 0 but the spec says it starts with zero if not specified [1]

Hmm, interesting I didn't know that. I got in the habit of at least providing the 0 one because @petere suggested that in a few PRs.
